Formula/ueberzugpp.rb

```diff
@@ -0,0 +1,36 @@
+class Ueberzugpp < Formula
+  desc "Drop in replacement for ueberzug written in C++"
+  homepage "https://github.com/jstkdng/ueberzugpp"
+  url "https://github.com/jstkdng/ueberzugpp/archive/refs/tags/v2.8.0.tar.gz"
+  sha256 "3f7e21052c3c218c436b1d2d8aedc1f02573ad83046be00b05069bfdc29bf4e2"
+  license "GPL-3.0-or-later"
+
+  depends_on "cli11" => :build
+  depends_on "cmake" => :build
+  depends_on "cppzmq" => :build
+  depends_on "nlohmann-json" => :build
+  depends_on "pkg-config" => :build
+  depends_on "fmt"
+  depends_on "libsixel"
+  depends_on "openssl@3"
+  depends_on "spdlog"
+  depends_on "tbb"
+  depends_on "vips"
+  depends_on "zeromq"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build",
+      "-DCMAKE_BUILD_TYPE=Release", "-Wno-dev",
+      "-DCMAKE_INSTALL_PREFIX='#{ENV["prefix"]}'",
+      "-DENABLE_X11=OFF", "-DENABLE_OPENCV=OFF",
+      "-DOPENSSL_ROOT_DIR=#{ENV["HOMEBREW_PREFIX"]}/opt/openssl",
+      *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    bin.install "build/ueberzug" => "ueberzugpp"
+  end
+
+  test do
+    system "#{bin}/ueberzugpp", "-V"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is kinda hard, `ueberzug` needs to do some calculations on the current pseudoterminal, and whilst the test might run locally, it could fail in the CI unless `ioctl` works.

Reply:
You can use Ruby's `pty` module. `grep` for `require "pty"` for examples.

Reply:
`pty` works but not completely, it allows U++ to start but not to display images (`ioctl` returns bad values). I've updated the test to just print the log created after startup.

Reply:
hmmm, test works on linux but on macos it fails. Does homebrew replace $TMPDIR ?